packages/next/src/client/components/navigation.ts

```diff
@@ -34,7 +34,7 @@ import { ReadonlyURLSearchParams } from './navigation.react-server'
  *
  * Read more: [Next.js Docs: `useSearchParams`](https://nextjs.org/docs/app/api-reference/functions/use-search-params)
  */
-function useSearchParams(): ReadonlyURLSearchParams {
+export function useSearchParams(): ReadonlyURLSearchParams {
```

Comment:
TypeScript can not augment those types with the ones from `packages/next/navigation-types/compat/navigation.d.ts` when separate export declarations are used. It fails with `Overload signatures must all be exported or non-exported.`

Reply:
Why did this need to change now?

Reply:
Thanks for nudging me to look closer, this is an interesting finding.

- The `packages/next/navigation-types/compat/navigation.d.ts` file is included in the typescript program because there are e2e/integration tests that have `/// <reference types="next/navigation-types/compat/navigation" />`  in their `next-env.d.ts`. This will be [generated like that whenever there is both app dir and pages dir](https://github.com/wyattjoh/next.js/blob/16a0e5af3213c0078b3069c795bf2bdad202c896/test/unit/write-app-declarations.test.ts#L102-L111).
- The compiler error appeared only now because `tsc` didn't even do a full compilation before with the broken `BrowserInterface` class.
- The compiler error would also appear in a freshly created Next.js project that uses app dir and also has a pages route, **_if_** `skipLibCheck` wouldn't be enabled by default.
- Here's a repro: https://github.com/unstubbable/next-navigation-types

So to summarise, you could consider this an accidental bugfix.